[FIX JENKINS-33799] Enforce correct icon size in list view

kohsuke opened the pull request:
Fix the width & height of the icon via CSS.

Previously, we were relying on the actual image being served as a
correct size. This is fine for icons we ship in Jenkins, but when
plugins use avatars as icons, those images might not come in the correct
size.

@reviewbybees in particular @tfennelly

kohsuke committed:
[FIXED JENKINS-33799]

Fix the width & height of the icon via CSS.

Previously, we were relying on the actual image being served as a
correct size. This is fine for icons we ship in Jenkins, but when
plugins use avatars as icons, those images might not come in the correct
size.

reviewbybees commented:
This pull request originates from a [CloudBees](https://www.cloudbees.com/) employee. At CloudBees, we require that all pull requests be reviewed by other CloudBees employees before we seek to have the change accepted. If you want to learn more about our process please see [this explanation](https://github.com/reviewbybees/about#about-reviewbybees).

daniel-beck commented:
@kohsuke Why does this specifically target 2.0, when it fixes an issue with a 1.x compatible plugin? Note that I plan to merge master before declaring 2.0 branch RC-ready, so this would still get in that way as well.

daniel-beck commented on war/src/main/webapp/css/style.css:
```diff
@@ -1975,3 +1975,23 @@ body.main-panel-only #main-panel {
     }
 }
+
+IMG.icon-sm {
```
Uppercase tags seems very unusual.

oleg-nenashev replied:
+1

kohsuke replied:
You youngsters don't seem to know, but historically [canonical tag names have been upper case](https://www.w3.org/TR/html4/index/elements.html). In HTML5 the canonical tag names seem to have switched to [lower case](https://www.w3.org/TR/html5/index.html) though.

recena replied:
I agree with @daniel-beck.

@kohsuke We want Jenkins is a modern software, for that reason, we have to look to -at least- to the present :smile: 

From other's exprerience:

> The simplest way to mitigate any potential issues surrounding case sensitivity is to always use lowercase for everything in your markup and CSS, where possible. If that’s not possible, make sure the case you use is consistent between your CSS and your document markup.

http://reference.sitepoint.com/css/casesensitivity

kohsuke commented:
> Why does this specifically target 2.0, when it fixes an issue with a 1.x compatible plugin?

As you said it hardly doesn't matter either way at this point.

kohsuke committed:
Whatever that makes other people comfortable

jglick commented:
:bee:

kohsuke commented:
I'd really like @tfennelly to sign this off given that he added those CSS classes.

tfennelly commented on war/src/main/webapp/css/style.css:
```diff
@@ -1975,3 +1975,23 @@ body.main-panel-only #main-panel {
+    height: 32px;
+}
+
+img.icon-xl {
```
I'm sure this should be `icon-xlg` (i.e. include a `g` character at the end).

tfennelly replied:
Prob better not to spec the element name either (`img`) as this could easily be applied to a `span` etc if applying the icon as a background image via css. Not a biggie for now though.

recena replied:
Better if we use `<img />` instead of `<span />` with background images. In this way, we can use to attributes like `alt` or `title` to improve the accessibility.

kohsuke replied:
I was mainly just restricting this to contain the impact of the change, but I'll modify as you suggest.

recena commented:
:bee:

kohsuke committed:
[JENKINS-33799]

As suggested by Tom, removing the tag name restriction
and correcting the xlg class name.

See the Icon class for more details

kohsuke commented:
Test failures are unrelated.

tfennelly commented:
:bee:

recena commented:
It seems @kohsuke did something similar [here](https://github.com/jenkinsci/jenkins/commit/cf0e6c68903938f608382f4c08624c350698ee12). We have CSS classes duplicated. This needs a cleanup.
